torch/_inductor/pattern_matcher.py

```diff
@@ -259,9 +259,33 @@ def replace_by_example(
                 trace_fn = functools.partial(
                     fwd_only, run_functional_passes=run_functional_passes
                 )
-            replacement = trace_fn(
-                replacement_fn, torch.fx.map_arg(args, lambda arg: arg.meta["val"])
-            )
+            if len(self.nodes) == 1 and "arg_kwarg_vals" in self.nodes[0].meta:
```

Comment:
Hmm, i dont think we should use arg_kwarg_vals here. i think we should use whatever the most recent meta is. which is what the current case is.

Reply:
Fair enough. The reason I did it this way is because it makes dealing with auto_functionalized ra lot easier, but yeah I agree tracing with arg_kwarg_vals is not the right call here